File: Math/Python/projection_area_of_3d_shapes.py
class Solution:
    def projectionArea(self, grid: list[list[int]]) -> int:
        le = len(grid)
        top, front, rear = le * le, 0, 0
        for i in range(le):
            maxrear, maxfront = 0, 0
            for j in range(le):
                curRear, curFront = grid[j][i], grid[i][j]
                if (curRear > maxrear): maxrear = curRear
                if (curFront > maxfront): maxfront = curFront
                if (curFront == 0): top -= 1
            rear += maxrear
            front += maxfront
        return top + front + rear

File: Math/JS/projection_area_of_3d_shapes.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var projectionArea = function(grid) {
    const len = grid.length;
    let top = 0, front = 0, rear = 0;
    const maxRear = new Array(len).fill(0);

    for (let i = 0; i < len; i++) {
        let maxFront = 0;
        for (let j = 0; j < len; j++) {
            if (grid[i][j] > 0) top++;
            maxFront = Math.max(maxFront, grid[i][j]);
            maxRear[j] = Math.max(maxRear[j], grid[i][j]);
        }
        front += maxFront;
    }

    rear = maxRear.reduce((sum, val) => sum + val, 0);
    return top + front + rear;
};